src/hb-ot-metrics.cc

```diff
@@ -32,6 +32,19 @@
 #include "hb-ot-metrics.hh"
 #include "hb-ot-face.hh"
 
+
+static float
+_fix_ascender_descender (float value, hb_ot_metrics_t metrics_tag)
+{
+  if (metrics_tag == HB_OT_METRICS_HORIZONTAL_ASCENDER ||
+      metrics_tag == HB_OT_METRICS_VERTICAL_ASCENDER)
+    return fabs ((double) value);
+  if (metrics_tag == HB_OT_METRICS_HORIZONTAL_DESCENDER ||
+      metrics_tag == HB_OT_METRICS_VERTICAL_DESCENDER)
+    return -fabs ((double) value);
+  return value;
+}
```

Comment:
I am not sure if we like to expose this logic to API users but you said this once IIRC, otherwise possible to do it hb-ot-font

Reply:
Yes, we should.  If we are fixing font crap, we should do it consistently.